crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -86,14 +87,87 @@ impl Feature<api::SetupMandate, types::SetupMandateRequestData> for types::Setup
         ))
         .await?;
 
-        mandate::mandate_procedure(
-            state,
-            resp,
-            maybe_customer,
-            pm_id,
-            connector.merchant_connector_id.clone(),
-        )
-        .await
+        if let Some(mandate_id) =
+            self.request
+                .setup_mandate_details
+                .as_ref()
+                .and_then(|mandate_data| {
+                    mandate_data
+                        .mandate_type
+                        .as_ref()
+                        .and_then(|mandate_data_type| match mandate_data_type {
+                            data_models::mandates::MandateDataType::SingleUse(_)
+                            | data_models::mandates::MandateDataType::MultiUse(_) => None,
+                            data_models::mandates::MandateDataType::UpdateMandateId(mandate_id) => {
+                                Some(mandate_id)
+                            }
+                        })
+                })
+        {
+            let mandate = state
+                .store
+                .find_mandate_by_merchant_id_mandate_id(&merchant_account.merchant_id, mandate_id)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MandateNotFound)?;
+
+            let profile_id = mandate::helpers::get_profile_id_for_mandate(
+                state,
+                merchant_account,
+                mandate.clone(),
+            )
+            .await?;
+            let connector_integration: services::BoxedConnectorIntegration<
+                '_,
+                types::api::MandateRevoke,
+                types::MandateRevokeRequestData,
+                types::MandateRevokeResponseData,
+            > = connector.connector.get_connector_integration();
+            let merchant_connector_account = helpers::get_merchant_connector_account(
```

Comment:
Since we don't have explicit validation for the profile id, I'm assuming this function takes care of it either explicitly or implicitly via a database call?

Reply:
Yes , this function gets the PaymentIntent for that `payment_id` and checks if there's a profile_id present for that PaymentInent